xlators/mgmt/glusterd/src/glusterd-handshake.c

```diff
@@ -1062,6 +1062,7 @@ __server_getspec(rpcsvc_request_t *req)
         if (ret < 0) {
             gf_msg("glusterd", GF_LOG_ERROR, errno, GD_MSG_FILE_OP_FAILED,
                    "Unable to stat %s (%s)", filename, strerror(errno));
+            op_errno = errno;
```

Comment:
Along with this error set check please put a condition check to set rsp.op_errno if it is not set
if (rsp.op_ret < 0) {
        gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_MOUNT_REQ_FAIL,
               "Failed to mount the volume");
        if (!op_errno)
           op_errno = ENOENT;
}

Reply:
Thanks for your help.

Reply:
@zhangxyue Can you resolve clang-error and resend a patch.